Userland/Libraries/LibCore/AnonymousBufferWindows.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, Andreas Kling <andreas@ladybird.org>
+ * Copyright (c) 2024, stasoid <stasoid@yahoo.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/AnonymousBuffer.h>
+#include <windows.h>
+
+namespace Core {
+
+AnonymousBufferImpl::AnonymousBufferImpl(AnonymousBufferId id, void* map_handle, size_t size, void* data)
+    : m_id(id)
+    , m_map_handle(map_handle)
+    , m_size(size)
+    , m_data(data)
+{
+}
+
+AnonymousBufferImpl::~AnonymousBufferImpl()
+{
+    if (m_data)
+        UnmapViewOfFile(m_data);
+
+    if (m_map_handle)
+        CloseHandle(m_map_handle);
+}
+
+ErrorOr<NonnullRefPtr<AnonymousBufferImpl>> AnonymousBufferImpl::create(size_t size)
+{
+    static int next_id_tag;
+    ByteString name = ByteString::formatted("AB-{}-{}", GetCurrentProcessId(), next_id_tag++);
+
+    HANDLE map_handle = CreateFileMapping(INVALID_HANDLE_VALUE, NULL, PAGE_READWRITE,
+        HIWORD(size), LOWORD(size), name.characters());
```

Comment:
This formatting looks off, it shouldn't be wrapped like this when using our clang-format-18 config. single-line is fine.